src/test/CMakeLists.txt

```diff
@@ -244,15 +249,43 @@ set(
     ${FILESYSTEM_LIBRARY}
     ${Boost_SYSTEM_LIBRARY}
     ${Boost_THREAD_LIBRARY}
+    ${CMAKE_DL_LIBS}
 )
 
 # Build special sanitizer version of googletest
 include_directories(../../third_party/googletest/googletest/)
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 
+# Configure TestPlugin
+add_library(TestPlugin SHARED
+    utils/test_plugin.cpp
+    utils/test_plugin.hpp
+)
+# Prevent the linker under macOS from complaining about undefined methods
+if (APPLE)
```

Comment:
Is that something all Plugins need to do? then maybe make it a macro/function, maybe fused with add_library. Could be called `add_hyrise_plugin` or something

Reply:
As I understand it, this has to be a setting of the plugin, yes. This actually brings up a very important point: I think the test plugin should be as stand-alone as all future plugins (except for not being in a different repository). Right now, it shares a lot of cmake "code" with the main code base.

Reply:
@mrzzzrm Yes, it is a Mac OS specific flag every plugin needs and I could encapsulate this with a function.

@mrks What exactly do you mean with a lot of make code? And how would we make it stand-alone? I could move the two test plugins to a specific plugin folder and they could have their own CMake file. Is that what you mean? I don't know if it's worth it. The plugin folder should then probably live inside the test folder because these plugins should only be used for testing purposes.

Reply:
In the future, actual plugins are supposed to live in their own repo. Thus, that repo should be able to be compiled on its own (outside of the Hyrise build process), correct?

Right now, the Hyrise build process builds these test plugins, so the way they are built does not reflect how actual plugins would be used. My idea would be to put them into a folder that looks like one of those plugin repos, and build them "from the outside". Does that make any sense?

Reply:
> The question is whether the build process of the test plugins should reflect the build process of actual plugins.

I would say yes, unless you have a good reason why they shouldn't.

Looking at your example repo, I wonder if the main part of the cmake file should be in the Hyrise repo so that the plugins only include it from the hyrise submodule. That way, whenever we make changes to the build process, we only have to make them in the main repo.